internal/runner/service.go

```diff
@@ -272,6 +274,13 @@ func (r *runnerService) Execute(srv runnerv1.RunnerService_ExecuteServer) error
 					return
 				}
 			}
+
+			// only update winsize when field is explicitly set
+			if req.ProtoReflect().Has(
```

Comment:
Alternatively, you could omit `optional` in the proto definition and validate the `Winsize` message, that is check if all fields are greater than zero (unless that's a valid scenario but I doubt it). If it's `nil` or a zero value, then `pty.Winsize{Cols: 80, Rows: 24}` would be the "normal" size. Some applications actually might expect 80 columns and 24 rows a minimal size (to double check).

Reply:
The reason I need it as optional is so that I'm not calling `pty.Setsize` when the user doesn't want to change the size. Otherwise the size would be set every time the user sends input, no?

Reply:
That's correct unless you cache the size you set at the startup/most recently and do not change it if the new size is the same (if it's a zero value = the default size).